packages/celotool/src/lib/geth.ts

```diff
@@ -486,23 +496,66 @@ export const transferCeloDollars = async (
     feeCurrency: txOptions.feeCurrency,
     gatewayFeeRecipient: txOptions.gatewayFeeRecipient,
     gatewayFee: txOptions.gatewayFee,
+    nonce: txOptions.nonce,
   })
 }
 
+export const unlock = async (
+  kit: ContractKit,
+  address: string,
+  password: string,
+  unlockPeriod: number,
+  unlockNeeded: boolean
+) => {
+  if (unlockNeeded) {
+    try {
+      kit.web3.eth.personal.unlockAccount(address, password, unlockPeriod)
+    } catch (error) {
+      console.error(`Unlock account ${address} failed:`, error)
+    }
+  }
+}
+
 export const simulateClient = async (
   senderAddress: string,
   recipientAddress: string,
   txPeriodMs: number, // time between new transactions in ms
   blockscoutUrl: string,
   blockscoutMeasurePercent: number, // percent of time in range [0, 100] to measure blockscout for a tx
-  index: number
+  index: number,
+  thread: number,
+  web3Provider: string = 'http://localhost:8545'
 ) => {
   // Assume the node is accessible via localhost with senderAddress unlocked
-  const kit = newKitFromWeb3(new Web3('http://localhost:8545'))
+  const kit = newKitFromWeb3(new Web3(web3Provider))
+  const password = fetchEnv('PASSWORD')
+
+  let lastNonce: number = 0
```

Comment:
I think the value representing "no last nonce" should be -1 instead of 0.  That way, if it's a new account, and the nonce to use is 0, `nonce` and `lastNonce` won't be equal.  See also comments below about how to get `nonce`.